aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -1082,17 +1082,26 @@ std::tuple<Tensor, Tensor> pack_hidden<std::tuple<Tensor, Tensor>>(const Tensor&
 }
 
 struct DropoutState {
+  // Both buffer and event are lazily instantiated when a dropout state is needed
+  // for the first time. Note that in this case needed != used, as we don't need
+  // a bufer to e.g. run RNNs in test mode.
   at::Tensor buffer;
-  cuda::CUDAEvent event;
+  at::optional<cuda::CUDAEvent> event;
```

Comment:
I'm impressed that Windows doesn't choke over this. I guess none of the cuDNN bindings need to be written in cu files :)